src/VCR/Videorecorder.php

```diff
@@ -216,7 +216,10 @@ public function handleRequest(Request $request)
             );
         }
 
+        $this->getMatchObserver()->startObserving();
+
         if ($this->cassette->hasResponse($request)) {
+            $this->getMatchObserver()->stopObserving();
```

Comment:
I have the feeling that `hasResponse` and `playback` are conceptually something else than finding out why a request doesn't match. Now it's doing more than one thing (see 'single responsibility principle'). 

How about introducing another method on the `Cassette` which does what you want and is named like that. Like `getMismatchReport()` or `getMismatchInfo()`. I imagine it would return an object like the `MatchObserver` which can be asked for a `getMismatchMessage()`.